pre-receive-hooks/block_self_merge_prs.sh

```diff
@@ -8,10 +8,9 @@
 # https://help.github.com/enterprise/admin/guides/developer-workflow/managing-pre-receive-hooks-on-the-github-enterprise-appliance/
 #
 
-if [ "$GITHUB_VIA" = "merge api" && "$GITHUB_PULL_REQUEST_AUTHOR_LOGIN" = "$GITHUB_USER_LOGIN"]; then
+if [[ "$GITHUB_VIA" = "merge api" ]] && [[ "$GITHUB_PULL_REQUEST_AUTHOR_LOGIN" = "$GITHUB_USER_LOGIN" ]]; then
```

Comment:
> is the `$GITHUB_VIA = "merge api"` doing enough here?

It was when i was writing this sample script, but agree totally that there are many other ways to merge that this does not cover. I'm :+1: to increasing it to handle some of those other merge methods.

> How would we prevent command-line merge commits from being pushed up? Would we still need to use protected branches?

That's a very good question. Basically we'd have to combine it with a pre-rx hook that restricts master commits that aren't a PR-backing